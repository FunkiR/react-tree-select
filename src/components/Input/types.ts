export type OnChangeEvent = {
	name: string,
	value: string
};

export type Props = {
    className?: string,
	name?: string,
    onChange: (event: OnChangeEvent) => void,
    value: string
};
