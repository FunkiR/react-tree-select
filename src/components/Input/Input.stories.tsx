import {action} from "@storybook/addon-actions";
import {useArgs} from '@storybook/client-api';
import {Meta, Story} from '@storybook/react/types-6-0';
import React from 'react';

import Input from './Input';
import type {Props} from './types';

export default {
	title: 'Components/Input',
	component: Input
} as Meta;

const Template: Story<Props> = args => {
	const [{value}, updateArgs] = useArgs();
	const onChange = (event) => {
		action('onChange')(event);
		updateArgs({value: event.value});
	};

	return <Input {...args} onChange={onChange} value={value} />;
};

export const Default = Template.bind({});

Default.args = {
	value: ''
};

