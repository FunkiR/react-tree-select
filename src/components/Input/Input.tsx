import React, {ChangeEvent, FC, memo, useState} from 'react';

import type {Props} from './types';

const Input: FC<Props> = props => {
	const {className, name = '', onChange, value: initValue} = props;
	const [value, setValue] = useState(initValue);

	const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
		const {value} = e.currentTarget;

		setValue(value);
		onChange({name, value});
	};

	return <input className={className} onChange={handleChange} value={value} />;
};

export default memo(Input);
