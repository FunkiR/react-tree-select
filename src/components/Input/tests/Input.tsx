import Input from "components/Input";
import {shallow} from "enzyme";
import React from "react";

describe('Input component', () => {
	it('renders component', () => {
		const wrapper = shallow(<Input onChange={jest.fn()} value={''} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('handles onChange and calls provided callback with event contains value', () => {
		const mockOnChange = jest.fn();
		const wrapper = shallow(<Input onChange={mockOnChange} value={''} />);
		const newValue = 'test';

		wrapper.find('input').simulate('change', {
			currentTarget: {
				value: newValue
			}
		});

		expect(mockOnChange.mock.calls.length).toEqual(1);
		expect(mockOnChange.mock.calls[0][0].value).toEqual(newValue);
	});
});
