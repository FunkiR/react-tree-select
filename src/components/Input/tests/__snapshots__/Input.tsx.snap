// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders component 1`] = `
<input
  onChange={[Function]}
  value=""
/>
`;

exports[`Input component renders correctly 1`] = `
<input
  onChange={[Function]}
  value=""
/>
`;
